lib/js/atoms/SurveyToggle/SurveyToggle.vue

```diff
@@ -0,0 +1,296 @@
+		background-color: $color-primary-disabled;
+	}
+
+	&.-selectedNeutral.-disabled &__toggle {
+		background-color: mix($color-minor-supporting, $color-total-white, 40%);
+	}
+
+	&.-selectedPrimary.-hovered &__toggle {
+		background-color: $color-primary-hovered;
+	}
+
+	&.-defaultPrimary.-hovered &__toggle {
+		background-color: $color-primary-background;
+	}
+
+	&.-selectedNeutral.-hovered &__toggle {
+		background-color: mix($color-minor-supporting, $color-firefly-black, 88%);
+	}
+
+	// ring
+	&__ring {
+		// TODO: IT-4131 mixin
+		align-items: center;
+		display: flex;
+		justify-content: center;
+		border: 3px solid $color-minor-supporting;
+		border-radius: 100%;
+		transition: border-color ease-in-out $default-transition-time;
+		width: 100%;
+	}
+
+	&.-disabled {
+		pointer-events: none;
+
+		#{$self}__ring {
+			border-color: mix($color-minor-supporting, $color-total-white, 40%);
+		}
+	}
+
+	&.-defaultPrimary &__ring {
+		border-color: $color-primary;
+	}
+
+	&.-selectedPrimary &__ring {
+		border-color: $color-total-white;
+	}
+
+	&.-selectedNeutral &__ring {
+		border-color: $color-total-white;
+	}
+
+	&.-defaultPrimary.-disabled &__ring {
+		border-color: $color-primary-disabled;
+	}
+
+	// content
+	&__content {
+		@include headlineXS();
+		@include textBold();
+
+		color: $color-minor-supporting;
+	}
+
+	&.-defaultPrimary &__content {
+		color: $color-primary;
+
+		.-hovered & {
+			$border-color: $color-primary-hovered;
+		}
+	}
+
+	&.-defaultPrimary.-disabled &__content {
+		color: $color-primary-disabled;
+	}
+
+	// icon
+	&__icon {
+		color: $color-total-white;
+		height: $icon-xs;
+		width: $icon-xs;
+	}
+
+	// label
+	&__label {
+		@include textInfoS();
+		@include textBold();
+
+		color: $color-minor;
+		text-align: center;
+		margin-top: $space-xxs;
+		min-height: 2em;
+		max-width: 100%;
+
+		@media #{breakpoint-s()} {
+			@include textInfoM();
+
+			font-weight: normal;
+		}
+	}
+}
+</style>
+
+<script lang="ts">
+import {
+	SURVEY_TOGGLE_MEANINGS,
+	SURVEY_TOGGLE_STATES,
+	SURVEY_TOGGLE_STATUSES,
+} from './SurveyToggle.consts';
+import DsIcon, { ICON_SIZES, ICONS } from '../Icon';
+import Ripple from 'vue-ripple-directive';
+
+export default {
+	name: 'SurveyToggle',
+	components: {
+		DsIcon,
+	},
+	directives: {
+		ripple: Ripple,
+	},
+	props: {
+		label: {
+			type: String,
+			default: null,
+		},
+		contentText: {
+			type: String,
+			default: null,
+		},
+		meaning: {
+			type: String,
+			default: SURVEY_TOGGLE_MEANINGS.PRIMARY,
+			validate(meaning) {
+				return Object.values(SURVEY_TOGGLE_MEANINGS).includes(meaning);
+			},
+		},
+		status: {
+			type: String,
+			default: SURVEY_TOGGLE_STATUSES.DEFAULT,
+			validate(status) {
+				return Object.values(SURVEY_TOGGLE_STATUSES).includes(status);
+			},
+		},
+		state: {
+			type: String,
+			default: SURVEY_TOGGLE_STATES.DEFAULT,
+			validate(state) {
+				return Object.values(SURVEY_TOGGLE_STATES).includes(state);
+			},
+		},
+		contentIcon: {
```

Comment:
Shouldn't this be called `selectedIcon`?

Reply:
Asked UX